cluster/images/provider-terraform/Dockerfile

```diff
@@ -1,3 +1,15 @@
-FROM BASEIMAGE
+FROM hashicorp/terraform:1.2.3
 
-COPY package.yaml .
+ARG TARGETOS
+ARG TARGETARCH
+
+ADD bin/$TARGETOS\_$TARGETARCH/provider /usr/local/bin/crossplane-terraform-provider
+ADD .gitconfig .gitconfig
+
+# As of Crossplane v1.3.0 provider controllers run as UID 2000.
+# https://github.com/crossplane/crossplane/blob/v1.3.0/internal/controller/pkg/revision/deployment.go#L32
+RUN mkdir /tf
+RUN chown 2000 /tf
```

Comment:
Hmm yeah this is kind of unfortunate the Crossplane decisions here are leaking into this image construction -- I'm going to do some more thinking about this but don't need to block 👍🏻

Reply:
It would be nice if we could mount the terraform working directory in the deployment/pod as a PVC so that the content could be saved across controller reboots.  We could put it somewhere that doesn't require the /tf directory to be created in the Dockerfile.

Reply:
@bobh66 ack -- linked this comment to https://github.com/crossplane/crossplane/issues/2535